### PR TITLE
PLAT-395 Standard Configuration of Programs

### PR DESCRIPTION
**Pull request by thees-k**
- Add method getDefaultCronExpression() to IProgram
- ProgramStandardConfiguration registers default schedule of programs, too
- Implement getDefaultCronExpression() for all programs of platform
- Delete WorkflowProgramStandardConfiguration because not necessary anymore

### CODE REVIEW

File: platform-core-module/src/main/java/com/softicar/platform/core/module/standard/configuration/ProgramStandardConfiguration.java [hunk 1]
```diff
@@ -16,14 +21,29 @@ public void createAndSaveAll() {
...
 		AGProgram program = new AGProgram()//
 			.setProgramUuid(uuid.getUuid())
 			.save();
 		program.getState().save();
+		registerDefaultSchedule(uuid);
+	}
+
+	private void registerDefaultSchedule(AGUuid uuid) {
+
+		Optional<IEmfSourceCodeReferencePoint> point = EmfSourceCodeReferencePoints.getReferencePoint(uuid.getUuid());
```
Comment (oliver-richers) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/standard/configuration/ProgramStandardConfiguration.java:38]: better use `getReferencePointOrThrow(IProgram.class, uud)`

**Comment by thees-k** [comment]
PLAT-395

File: platform-core-module/src/main/java/com/softicar/platform/core/module/program/IProgram.java [hunk 2]
```diff
@@ -12,4 +12,9 @@ default IDisplayString toDisplay() {
 	}
 
 	void executeProgram();
+
+	default String getDefaultCronExpression() {
```
Comment (oliver-richers) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/program/IProgram.java:16]: should provide a Javadoc comment, especially concerning the possible results, e.g. `""`, `null`, etc.

**Review by oliver-richers — CHANGES_REQUESTED** [review]

File: platform-core-module/src/main/java/com/softicar/platform/core/module/program/IProgram.java [hunk 3]
```diff
@@ -12,4 +14,16 @@ default IDisplayString toDisplay() {
 	}
 
 	void executeProgram();
+
+	/**
+	 * This method can be overwritten in that way that it returns an Optional
+	 * with a cron expression. In this case the program will be scheduled
+	 * automatically with it (see {@link ProgramStandardConfiguration}).
+	 *
+	 * @return An Optional with a cron expression, or an empty Optional.
```
Comment (oliver-richers) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/program/IProgram.java:23]:
```suggestion
	 * @return an optional cron expression
```

**Review by oliver-richers — CHANGES_REQUESTED** [review]

**Review by oliver-richers — APPROVED** [review]

**Pull request merged**